src/lib/controller/project-controller.ts

```diff
@@ -0,0 +1,71 @@
+import type { Criterion, CriterionSpec, Project, ProjectSpec, User } from "$lib/model/backend";
+import type { ICriterionController, IProjectController, IStageController } from "../backend-api";
+import { CriterionController } from "./criterion-controller";
+import { HttpClient } from "./http-client";
+import { StageController } from "./stage-controller";
+
+export class ProjectController implements IProjectController {
+    private client: HttpClient;
+    private path: string;
```

Comment:
I think they can be readonly / final (same in other controllers too)?!

Reply:
Yes true, I changed that.